modules/azure/automationaccount.go

```diff
@@ -0,0 +1,342 @@
+	require.NoError(t, err)
+
+	return compiled
+}
+
+// AutomationAccountDscCompiledE indicates whether the specified Azure Automation Account DSC compiled successfully.
+// DSC compilation is performed via a Terraform null_resource using PowerShell Core, executing async.
+// Compilation can take a few minutes to spin up resources requiring a retry mechanism to allow for this
+func AutomationAccountDscCompiledE(t testing.TestingT, dscConfiguraitonName string, automationAccountName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	seconds := 30 // 30 second initial delay
+	sleep := time.Duration(seconds) * time.Second
+	maxAttempts := 5 // try 5 times, doubling the delay each time.
+
+	for i := 1; i < maxAttempts; i++ {
+		time.Sleep(sleep)
+		dscCompileJobStatus, err := AutomationAccountDscCompileJobStatusE(t, dscConfiguraitonName, automationAccountName, resourceGroupName, subscriptionID)
+		if err != nil {
+			return false, err
+		}
+		// check if status === Completed
+		if dscCompileJobStatus == "Completed" {
+			return true, nil
+		}
+		sleep = 2 * sleep
+	}
+	return false, nil
+}
+
+// AutomationAccountRunAsCertificateThumbprintMatches indicates whether the specified Azure Automation Account RunAs Certificate exists.
+// This function would fail the test if there is an error.
+func AutomationAccountRunAsCertificateThumbprintMatches(t testing.TestingT, runAsCertificateThumbprint string, runAsCertificateName string, automationAccountName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := AutomationAccountRunAsCertificateThumbprintMatchesE(t, runAsCertificateThumbprint, runAsCertificateName, automationAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return exists
+}
+
+// AutomationAccountRunAsCertificateThumbprintMatchesE indicates whether the specified Azure Automation Account RunAs Certificate exists.
+func AutomationAccountRunAsCertificateThumbprintMatchesE(t testing.TestingT, runAsCertificateThumbprint string, runAsCertificateName string, automationAccountName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	certificate, err := GetAutomationAccountCertificateE(t, runAsCertificateName, automationAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return *certificate.CertificateProperties.Thumbprint == runAsCertificateThumbprint, nil
+}
+
+// AutomationAccountRunAsConnectionValidates indicates whether the specified Azure Automation Account RunAs Account exists.
+// This function would fail the test if there is an error.
+func AutomationAccountRunAsConnectionValidates(t testing.TestingT, automationAccountrunAsAccountName string, runAsConnectionType string, runAsCertificateThumbprint string, automationAccountName string, resourceGroupName string, subscriptionID string) bool {
```

Comment:
name is confusing!, is there a more accurate name we can use, may be AutomationAccountHasARunAsConnection

Reply:
Fixed in latest commit.